gammapy/cube/make.py

```diff
@@ -268,9 +275,30 @@ def run(self, selection=None):
         for name in selection:
             getattr(self, "_make_" + name)()
 
+        background_model = BackgroundModel(self.maps["background"])
+
+        if "psf" in self.maps:
+            psf = self.maps["psf"]
+        else:
+            psf = None
+
+        if "edisp" in self.maps:
+            edisp = self.maps["edisp"]
+        else:
+            edisp = None
+
+        dataset = MapDataset(
+            counts=self.maps["counts"],
+            exposure=self.maps["exposure"],
+            background_model=background_model,
+            psf=psf,
+            edisp=edisp,
```

Comment:
As there is the access to the `self.observation` objects, you could also copy over the `GTI` table here and maybe set the name of the dataset to `"obs-{}".format(observation.obs_id)` or similar.

Reply:
Thanks @adonath ! Addressed the changes...